### PR TITLE
Helm Chart Update

### PR DESCRIPTION
**Pull request by ajcrowe**
This updates the Helm chart with a few extra params and simplifies some of the common configuration

* Updated to use dedicated resources and pod lables/annotations
* Added ConfigMap for common envs
* Updated storage configuration to be more flexible
* Added missing envs for db path, backtrace and loggings

### CODE REVIEW

**Comment by yourbuddyconner** [comment]
Tested this change and it works great, thanks for the contribution @ajcrowe!

**Review by yourbuddyconner — APPROVED** [review]

File: rust/helm/optics-agent/templates/_helpers.tpl [hunk 1]
```diff
@@ -35,7 +35,6 @@ Common labels
 */}}
 {{- define "optics-agent.labels" -}}
 helm.sh/chart: {{ include "optics-agent.chart" . }}
-optics/deployment: {{ .Values.optics.runEnv | quote }}
```
Comment (yourbuddyconner) [inline rust/helm/optics-agent/templates/_helpers.tpl:38]: @ajcrowe We actually use this label in our dashboards to filter logs from a particular deployment. Do you have any suggestions on how we can preserve this functionality?

**Comment by prestwich** [comment]
failing checks are because of recent changes to CI workflows. Will rerun

**Comment by prestwich** [comment]
ah correction, checks are failing due to #624

**Pull request merged**